configure.ac

```diff
@@ -717,13 +726,19 @@ AC_SUBST(OLA_CLIENT_LIBS)
 
 # Extra tools
 #####################################################
-AM_CONDITIONAL([BUILD_JA_RULE],
-               [test "x$enable_ja_rule" = xyes && test "x$have_libusb" = xyes \
-                && test "x$have_libusb_hotplug_api" = xyes])
+AS_IF([test "x$enable_ja_rule" = xyes && test "x$enable_usbdmx" = xno],
+      [AC_MSG_ERROR([Ja Rule requires the usbdmx plugin, but it could not be enabled.])])
+AS_IF([test "x$enable_ja_rule" = xyes && test "x$have_libusb" = xno],
+      [AC_MSG_ERROR([Ja Rule requires libusb, but it was not found on your system.])])
+AS_IF([test "x$enable_ja_rule" = xyes && test "x$have_libusb_hotplug_api" = xno],
+      [AC_MSG_ERROR([Ja Rule requires libusb to have the hotplug capability (versions 1.0.16 \
+and above), but the version found on your system doesn't appear to have it.])])
+
+AM_CONDITIONAL([BUILD_JA_RULE], [test "x$enable_ja_rule" = xyes])
```

Comment:
AC_MSG_ERROR prints the error and exits the script with failure.

Currently, the default value for Ja Rule is that it doesn't get built (therefore it doesn't have an auto mode). You just enable it if you want it. If you enable it and a dependency is missing, configure stops and tells you there was a problem.

If we want Ja Rule to be auto by default, then we'll need to refactor this so that if a condition fails, we either fail the configure script or silently disable the feature, depending on the input state.